examples/noline.rs

```diff
@@ -0,0 +1,248 @@
+extern crate menu;
+
+use embedded_io::{ErrorType, Read as EmbRead, Write as EmbWrite};
+use menu::*;
+use noline::builder::EditorBuilder;
+use std::{
+    io::{self, Read as _, Stdin, Stdout, Write as _},
+    iter::repeat,
+};
+use termion::raw::IntoRawMode;
+
+pub struct IOWrapper {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl IOWrapper {
+    pub fn new() -> Self {
+        Self {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+}
+
+impl Default for IOWrapper {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ErrorType for IOWrapper {
+    type Error = embedded_io::ErrorKind;
+}
+
+impl EmbRead for IOWrapper {
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize, Self::Error> {
+        self.stdin
+            .read_exact(&mut buf[0..1])
+            .map_err(|e| e.kind())?;
+        Ok(1)
+    }
+}
+
+impl EmbWrite for IOWrapper {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, Self::Error> {
+        let mut written = 0;
+        let parts = buf.split(|b| *b == b'\n').collect::<Vec<_>>();
+
+        for (i, part) in parts.iter().enumerate() {
+            written += self.stdout.write(part).map_err(|e| e.kind())?;
+
+            if i != parts.len() - 1 {
+                let _ = self.stdout.write(b"\n\r").map_err(|e| e.kind())?;
```

Comment:
Should this be `\r\n`?

Reply:
Noline uses `\r\n` internally since it deals with raw terminals and uart. I noticed that menu uses `\n`. How does that work with raw terminals? Do you need to add a `\r` everytime you encounter an `\n` in `write()`?

Reply:
Yeah I was relying on the terminal being cooked. If you're in raw mode you want to send both I think.

But you wrote `\n\r` which is reversed.

Reply:
Ah, ok, I didn't even noticed that they were reversed :)